src/app/message/message.component.ts

```diff
@@ -0,0 +1,219 @@
+import { Component, ChangeDetectorRef, OnInit, ViewChild, ElementRef, HostListener} from '@angular/core';
+import { MessageApiService } from '../message-api.service';
+import { faMicrophone, faPaperPlane, faEraser, faArrowDown, faFile} from '@fortawesome/free-solid-svg-icons';
+declare var webkitSpeechRecognition: any;
+declare var speechSynthesis: any;
+
+@Component({
+  selector: 'app-message',
+  templateUrl: './message.component.html',
+  styleUrls: ['./message.component.css'],
+})
+export class MessageComponent implements OnInit {
+  allMessages: any[] = [];
+  newMessage: string = '';
+  loginInput: string = '';
+  passwordInput: string = '';
+  voiceMessage: string = '';
+  showScrollButton: boolean = false; 
+  isRecording: boolean = false; 
+  isTyping: boolean = false;
+  iconMicrophone = faMicrophone;
+  iconSend = faPaperPlane;
+  iconClear = faEraser;
+  iconSkipToNew = faArrowDown;
+  iconNew = faFile;
+  isLoginPopupOpen: boolean = false;
+  isLoginSuccessful: boolean = false;
+  isLoginError: boolean = false;
+
+
+  @ViewChild('messageContainer') private messageContainer?: ElementRef;
+
+  constructor(private messageApiService: MessageApiService,
+              private cdr: ChangeDetectorRef
+              ) {}
+
+  ngOnInit() {
+    this.loadMessage();
+    this.isLoginPopupOpen = true;
+  }
+
+  loadMessage() {
+    this.messageApiService.getMessage().subscribe(
+      (response) => {
+        this.allMessages = response.messages;
+        this.readLastAssistantMessage();
+      },
+      (error) => {
+        console.error('Error loading messages:', error);
```

Comment:
Brak porzadnej obsługi błędów - zobacz jak to jest w planner lite dotyczy to wielu miejsc.